frontend/packages/integration-tests-cypress/tests/crud/k8-openshift-cruds.cyspec.ts

```diff
@@ -0,0 +1,190 @@
+    .set('resourcequotas', { kind: 'ResourceQuota' })
+    .set('limitranges', { kind: 'LimitRange' })
+    .set('horizontalpodautoscalers', { kind: 'HorizontalPodAutoscaler' })
+    .set('networkpolicies', { kind: 'NetworkPolicy' })
+    .set('roles', { kind: 'Role' });
+  const openshiftObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>()
+    .set('deploymentconfigs', { kind: 'DeploymentConfig' })
+    .set('buildconfigs', { kind: 'BuildConfig' })
+    .set('imagestreams', { kind: 'ImageStream' })
+    .set('routes', { kind: 'Route' })
+    .set('user.openshift.io~v1~Group', { kind: 'user.openshift.io~v1~Group', namespaced: false });
+  const serviceCatalogObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>().set(
+    'clusterservicebrokers',
+    {
+      kind: 'servicecatalog.k8s.io~v1beta1~ClusterServiceBroker',
+      namespaced: false,
+    },
+  );
+
+  let testObjs = Cypress.env('openshift') === true ? k8sObjs.merge(openshiftObjs) : k8sObjs;
+  testObjs = Cypress.env('servicecatalog') === true ? testObjs.merge(serviceCatalogObjs) : testObjs;
+  const testLabel = 'automatedTestName';
+  const resourcesWithCreationForm = new Set(['StorageClass', 'Route', 'PersistentVolumeClaim']);
+  const leakedResources = new Set<string>();
+
+  testObjs.forEach(({ kind, namespaced = true }, resource) => {
+    describe(kind, () => {
+      const name = `${testName}-${_.kebabCase(kind)}`; // end name with '&' to force test failures
+      it('displays a list view for the resource', () => {
+        cy.visit(
+          `${namespaced ? `/k8s/ns/${testName}` : '/k8s/cluster'}/${resource}?name=${testName}`,
+        );
+      });
+      if (namespaced) {
+        it('has a working namespace dropdown on namespaced objects', () => {
+          listPage.projectDropdownShouldExist();
+          listPage.projectDropdownShouldContain(testName);
+        });
+      } else {
+        it('does not have a namespace dropdown on non-namespaced objects', () => {
+          listPage.projectDropdownShouldNotExist();
+        });
+      }
+      it('creates, searches for, edits, and deletes a resource instance', () => {
+        listPage.clickCreateYAMLButton();
+        if (resourcesWithCreationForm.has(kind)) {
+          cy.get('#yaml-link').click();
+        }
+        // for 'create yaml' resource sidebar is rendered last
+        cy.byTestID('resource-sidebar').should('exist');
+        let newContent;
+        // get, update, and set yaml editor content.
+        yamlEditor
+          .getEditorContent()
+          .then((content) => {
+            newContent = _.defaultsDeep(
+              {},
+              { metadata: { name, labels: { [testLabel]: testName } } },
+              safeLoad(content),
+            );
+          })
+          .then(() => {
+            yamlEditor.setEditorContent(safeDump(newContent)).then(() => {
+              cy.log('creates a new resource instance');
+              leakedResources.add(
```

Comment:
Since we're deleting the project when done, this isn't as important. We might be able to skip the leaked resources logic, or only track cluster-scoped resources.

Reply:
ok, was wondering if this was needed in new Cypress test env.  Removed the leaked resources logic.